src/effects/VST3/VST3Utils.cpp

```diff
@@ -52,124 +52,145 @@ bool VST3Utils::ParsePluginPath(const wxString& pluginPath, wxString* modulePath
                );
-               combo->AppendString(VST3Utils::ToWxString(displayValue));
+               ctrlDouble->SetIncrement((ctrlDouble->GetMax() - ctrlDouble->GetMin()) * 0.01);
+               ctrlDouble->SetValue(editController.normalizedParamToPlain(parameterInfo.id, currentNormalizedValue));
+               
+               ctrlDouble->Bind(wxEVT_SPINCTRLDOUBLE, [this, id = parameterInfo.id](const wxSpinDoubleEvent& event) {
+                  UpdateParameter(id, event.GetValue());
+               });
+               sizer->Add(ctrlDouble, 0, wxEXPAND);
             }
-            
-            combo->SetSelection(
-               static_cast<int>(editController->normalizedParamToPlain(parameterInfo.id, currentNormalizedValue))
-            );
-            combo->Bind(wxEVT_COMBOBOX, [updateParameter, id = parameterInfo.id](const wxCommandEvent& event) {
-               const auto comboBox = static_cast<wxComboBox*>(event.GetEventObject());
-               updateParameter(id, comboBox->GetSelection());
-            });
-            gridSizer->Add(combo, 0, wxEXPAND);
+            else if(parameterInfo.stepCount == 1)//toggle
+            {
+               const auto checkbox = safenew wxCheckBox (this, wxID_ANY, wxEmptyString);
+               checkbox->SetValue(currentNormalizedValue != .0);
+               checkbox->Bind(wxEVT_CHECKBOX, [this, id = parameterInfo.id](const wxCommandEvent& event){
+                  const auto checkbox = static_cast<wxCheckBox*>(event.GetEventObject());
+                  UpdateParameter(id, checkbox->GetValue() ? 1. : .0);
+               });
+               sizer->Add(checkbox, 0, wxEXPAND);
+            }
+            else if(parameterInfo.stepCount > 0)//discrete range
+            {
+               const auto maxValue = parameterInfo.stepCount;
+               if(parameterInfo.flags & Vst::ParameterInfo::kIsList)
+               {
+                  const auto choice = safenew wxChoice(this, wxID_ANY);
+
+                  for(auto j = 0; j <= maxValue; ++j)
+                  {
+                     Vst::String128 displayValue = { 0 };
+                     editController.getParamStringByValue(
+                        parameterInfo.id,
+                        editController.plainParamToNormalized(parameterInfo.id, j),
+                        displayValue
+                     );
+                     choice->AppendString(VST3Utils::ToWxString(displayValue));
+                  }
+                  
+                  choice->SetSelection(
+                     static_cast<int>(editController.normalizedParamToPlain(parameterInfo.id, currentNormalizedValue))
+                  );
+                  choice->Bind(wxEVT_CHOICE, [this, id = parameterInfo.id](const wxCommandEvent& event) {
+                     const auto comboBox = static_cast<wxComboBox*>(event.GetEventObject());
+                     UpdateParameter(id, comboBox->GetSelection());
```

Comment:
The first commit introduces the new window class but also changes the combo to choice, and introduces an error, because the `static_cast` above did not change the class.

Perhaps make one commit to change combo to choice first so that change isn't hidden among all the other diff for this commit.